### PR TITLE
Added avl_ndel() interface

### PR DESCRIPTION
**Pull request by matthiasbeyer**
Solves #110 .

### CODE REVIEW

**Comment by neithernut** [comment]
I would add a warning about functions not accepting `NULL`, but this may also be done in the implementation PR.
Otherwise: ACK

**Comment by matthiasbeyer** [comment]
Travis failes. We should fix it before merging!

File: src/libreset/avl.h [hunk 1]
```diff
@@ -122,6 +122,22 @@ avl_del(
...
+ *
+ * @memberof avl
+ *
+ * @reutrn the number of removed elements
+ */
+unsigned int
+avl_ndel(
+    struct avl* el, //!< The avl where to search in
+    struct r_set_cfg* cfg //!< type information provided by the user
+};
```
Comment (neithernut) [inline src/libreset/avl.h]: The build fails because of the typo in this line. It should be a `)`, not a `}`.

**Comment by neithernut** [comment]
Relates to issue #103.

**Comment by neithernut** [comment]
As already pointed out: ACK

**Comment by matthiasbeyer** [comment]
Waiting on CI.

**Pull request merged**